agent.go

```diff
@@ -146,4 +146,7 @@ type agent interface {
 	// container related to a Pod. If all is true, all processes in
 	// the container will be sent the signal.
 	killContainer(pod Pod, c Container, signal syscall.Signal, all bool) error
+
+	// psContainer will list the processes running inside the container
+	psContainer(pod Pod, c Container, format string, psArgs []string) ([]byte, error)
```

Comment:
`processListContainer` seems like a more generic name IMO.
And also I would prefer a function prototype a bit different:
```go
processListContainer(pod Pod, c Container, processListOptions []Param) (ProcessList, error)
```
The reason here is that I would like to get the same processList structure from any agent implementation and get this structure formatted as expected from the runtime implementation.